### ISSUE TITLE
Require Java 17

### ISSUE THREAD

**Issue by emilpaw** [jhipster/generator-jhipster#19768]
##### **Overview of the feature request**

Make Java 17 the new baseline and drop support for older versions.

##### **Motivation for or Use Case**

This will allow the usage of many new Java features and keep the Tech Stack of the generated code up to date. 

Although Java 11 has still one year of support it's already four years old. Java 17 was released over a year ago so at this point it's pretty safe to bet on it. 

Java 17 needs to be made the baseline at the latest when upgrading to Spring Boot 3 which will be released in two months. It might be worth considering adopting Java 17 before that.  

From the [Preparing for Spring Boot 3.0](https://spring.io/blog/2022/05/24/preparing-for-spring-boot-3-0) blog post:
> Spring Boot 3.0 will require Java 17, but you don’t need to wait until that release to upgrade to the latest LTS Java version. Any recent Spring Boot 2.x release will work really well with Java 17. You can also make use of Java 17 features (such as records) in your own codebases.
> We highly recommend that you upgrade your JDK today if at all possible.

##### **Related issues or PR**

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by mraible** [comment]
I don't think we should force Java 17 until we have Spring Boot 3 support. We can probably start that migration now since there are milestone releases. For example, M4:

https://github.com/spring-projects/spring-boot/releases/tag/v3.0.0-M4

**Comment by atomfrede** [comment]
Agree with Matt. We build and test with java 17, staying to java 11 feature. We will update to java 17 as minimum with upcoming v8 release cycle (and of course spring boot 3 will also force us to do).
